src/ai/backend/manager/models/endpoint.py

```diff
@@ -631,6 +645,87 @@ class Meta:
     items = graphene.List(Endpoint, required=True)
 
 
+class ModifyEndpointInput(graphene.InputObjectType):
+    resource_slots = graphene.JSONString()
+    resource_opts = graphene.JSONString()
+    cluster_mode = graphene.String()
+    cluster_size = graphene.Int()
+    desired_session_count = graphene.Int()
+    image = ImageRefType()
+    name = graphene.String()
+    resource_group = graphene.String()
+    open_to_public = graphene.Boolean()
+
+
+class ModifyEndpoint(graphene.Mutation):
+    class Arguments:
+        endpoint_id = graphene.UUID(required=True)
+        props = ModifyEndpointInput(required=True)
+
+    ok = graphene.Boolean()
+    msg = graphene.String()
+    endpoint = graphene.Field(lambda: Endpoint, required=False, description="Added since 23.09.9.")
+
+    @classmethod
+    async def mutate(
+        cls,
+        root,
+        info: graphene.ResolveInfo,
+        endpoint_id: uuid.UUID,
+        props: ModifyEndpointInput,
+    ) -> "ModifyEndpoint":
+        graph_ctx: GraphQueryContext = info.context
+        data: dict[str, Any] = {}
+        set_if_set(
+            props,
+            data,
+            "resource_slots",
+            clean_func=lambda v: ResourceSlot.from_user_input(v, None),
+        )
+        set_if_set(props, data, "resource_opts")
+        set_if_set(props, data, "cluster_mode")
+        set_if_set(props, data, "cluster_size")
+        set_if_set(props, data, "desired_session_count")
+        set_if_set(props, data, "image")
+        set_if_set(props, data, "resource_group")
+        image = data.pop("image", None)
+        if image is not None:
+            image_name = image["name"]
+            registry = image.get("registry") or ["*"]
+            arch = image.get("architecture")
+            if arch is not None:
+                image_ref = ImageRef(image_name, registry, arch)
+            else:
+                image_ref = ImageRef(
+                    image_name,
+                    registry,
+                )
+            async with graph_ctx.db.begin_readonly_session() as db_session:
+                image_row = await ImageRow.resolve(db_session, [image_ref])
+            data["image"] = image_row.id
+
+        update_query = sa.update(EndpointRow).values(**data).where(EndpointRow.id == endpoint_id)
+
+        async def _post(conn, result) -> EndpointRow:
+            db_session = AsyncSession(conn)
+            row = result.first()
+            query = (
+                sa.select(EndpointRow)
+                .where(EndpointRow.id == row.id)
+                .options(
+                    joinedload(EndpointRow.image_row),
+                    selectinload(EndpointRow.routings),
+                    joinedload(EndpointRow.session_owner_row),
+                    joinedload(EndpointRow.created_user_row),
+                )
+            )
```

Comment:
Please use `Endpoint.get()` ([ref](https://github.com/lablup/backend.ai/blob/0270a0fdbbdf3c6784f7c5d0d9dd3dd36c62ab34/src/ai/backend/manager/models/endpoint.py#L202-L240)) instead of directly constructing query.